tests/virt/node/general/utils.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from datetime import datetime
+
+from utilities.infra import run_virtctl_command
+from utilities.virt import running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+MARKER_PATH = "~/reset_marker"
+
+
+def record_vm_state_for_reset(vm):
+    return {
+        "launcher_pod_uid": vm.vmi.virt_launcher_pod.instance.metadata.uid,
+        "boot_time": get_vm_latest_boot_time(vm=vm),
+    }
+
+
+def verify_reset_occurred(vm, original_state):
+    current_launcher_pod_uid = vm.vmi.virt_launcher_pod.instance.metadata.uid
+    assert current_launcher_pod_uid == original_state["launcher_pod_uid"], (
+        f"Launcher pod UID changed after reset. "
+        f"Original: {original_state['launcher_pod_uid']}, "
+        f"Current: {current_launcher_pod_uid}"
+    )
+
+    assert verify_vm_rebooted(vm, original_state["boot_time"]), (
+        f"Guest reboot not detected. No newer boot entry found after reset. "
+        f"Original boot time: {original_state['boot_time']}"
+    )
+
+    LOGGER.info("Reset verified: same pod UID, guest rebooted with newer boot entry")
+
+
+def reset_vm(vm):
+    LOGGER.info(f"Resetting VM {vm.name}")
+
+    run_virtctl_command(command=["reset", vm.name], namespace=vm.namespace)
+
+    LOGGER.info(f"Waiting for VM {vm.name} to be ready after reset")
+    from utilities.constants import TIMEOUT_5MIN
+
+    running_vm(vm=vm, ssh_timeout=TIMEOUT_5MIN)
+
+    LOGGER.info(f"VM {vm.name} is ready after reset")
+    return True
+
+
+def get_vm_latest_boot_time(vm):
+    result = vm.ssh_exec.run_command(command=["last", "-F", "reboot", "-n", "1"])[1].strip()
+
+    parts = result.split()
+    date_str = " ".join(parts[4:9])
+    latest_boot_time = datetime.strptime(date_str, "%a %b %d %H:%M:%S %Y")
+
+    LOGGER.info(f"Latest boot time found: {latest_boot_time}")
+    return latest_boot_time
+
+
+def verify_vm_rebooted(vm, original_boot_time):
+    current_boot_time = get_vm_latest_boot_time(vm=vm)
+
+    if current_boot_time > original_boot_time:
+        LOGGER.info(f"VM reboot confirmed: {original_boot_time} -> {current_boot_time}")
+        return True
+    else:
+        LOGGER.warning(f"No reboot detected: {original_boot_time} vs {current_boot_time}")
+        return False
+
+
+def create_ephemeral_marker(vm):
+    result = vm.ssh_exec.run_command(command=["touch", MARKER_PATH])
+
+    if result[0] != 0:
+        raise RuntimeError(f"Failed to create marker file: {result[2]}")
```

Comment:
probably `run_ssh_commands()` can cover that, see example:

https://github.com/RedHatQE/openshift-virtualization-tests/blob/main/tests/virt/cluster/vm_cloning/test_vm_cloning.py#L95

may be you can do something similar